### DISCUSSION TITLE
compile the core to WebAssembly

### DISCUSSION THREAD

**Discussion by smallstepman** [supermerill/SuperSlicer#2946] [General]
How feasable it would be to compile this to webassembly (leaving GUI aside - i.e. compiling only the core/backend)?

This would setup a path which leads to having SuperSlicer embedded inside other web clients, like octoprint, fluidd, mainsail etc.

It would be the the best thing since sliced bread to have this running in the browser with some GUI written in JS.

**Comment by supermerill** [comment]
The `src/libslic3r` directory contains all the non-gui stuff, and it doessn't depends on the gui stuff that is in `src/slic3r`.
You can try yourself, good luck.

**Reply by smallstepman** [reply]
Thanks for pointing me in the right direction. I'm coming from Rust and I'm not very familiar with C++. Could advice where I'll find the entry point for `src/libslic3r`?

**Reply by supermerill** [reply]
https://github.com/supermerill/SuperSlicer/tree/master/src/libslic3r

you have to create a good configuration and print object, and then call process on that print object.

You can look at prusaslicer.cpp, on how it works with command-line.
